src/Exception/TransactionRolledBack.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Doctrine\DBAL\Exception;
+
+use Doctrine\DBAL\Driver\Exception as TheDriverException;
+use Doctrine\DBAL\Query;
+
+/** @psalm-immutable */
+class TransactionRolledBack extends DriverException
+{
+    private DriverException $why;
+
+    public function __construct(DriverException $why, TheDriverException $driverException, ?Query $query)
+    {
+        parent::__construct($driverException, $query);
+
+        $this->why = $why;
+    }
+
+    public function why(): DriverException
```

Comment:
Is this method called anywhere?

Reply:
Nope but I'd keep it as a public api to be able to fetch the cause of the exception

Reply:
I don't think users should have anything to do to see the exception. Maybe `$why` and `$exception` should be chained together, just like what happens when having exception in `try` and the corresponding `finally`?

Reply:
Good point, now I see it can be simplified.

Reply:
(the CI error does not seem relevant)